[pruner] amend base pruner API to match base sparsifier

kazhou opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63178
* #63158

Update base pruner API to match base sparsifier API as defined in D28970960 / PR58955

Changes include:
- `enable_mask_update = True` in `__init__`
- `prepare` takes model and config instead of constructor
- convert functionality renamed to `squash_mask`, `convert` method call now raises Error
- `activation_handles` ad `bias_handles` initialized in `_prepare` instead of constructor

Differential Revision: [D30287179](https://our.internmc.facebook.com/intern/diff/D30287179/)

kazhou committed:
[pruner] amend base pruner API to match base sparsifier

Update base pruner API to match base sparsifier API as defined in D28970960 / PR58955

Changes include:
- `enable_mask_update = True` in `__init__`
- `prepare` takes model and config instead of constructor
- convert functionality renamed to `squash_mask`, `convert` method call now raises Error
- `activation_handles` ad `bias_handles` initialized in `_prepare` instead of constructor

Differential Revision: [D30287179](https://our.internmc.facebook.com/intern/diff/D30287179/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] amend base pruner API to match base sparsifier"


Update base pruner API to match base sparsifier API as defined in D28970960 / PR58955

Changes include:
- `enable_mask_update = True` in `__init__`
- `prepare` takes model and config instead of constructor
- convert functionality renamed to `squash_mask`, `convert` method call now raises Error
- `activation_handles` ad `bias_handles` initialized in `_prepare` instead of constructor

Differential Revision: [D30287179](https://our.internmc.facebook.com/intern/diff/D30287179/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] amend base pruner API to match base sparsifier"


Update base pruner API to match base sparsifier API as defined in D28970960 / PR58955

Changes include:
- `enable_mask_update = True` in `__init__`
- `prepare` takes model and config instead of constructor
- convert functionality renamed to `squash_mask`, `convert` method call now raises Error
- `activation_handles` ad `bias_handles` initialized in `_prepare` instead of constructor

Differential Revision: [D30287179](https://our.internmc.facebook.com/intern/diff/D30287179/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] amend base pruner API to match base sparsifier"


Update base pruner API to match base sparsifier API as defined in D28970960 / PR58955

Changes include:
- `enable_mask_update = True` in `__init__`
- `prepare` takes model and config instead of constructor
- convert functionality renamed to `squash_mask`, `convert` method call now raises Error
- `activation_handles` ad `bias_handles` initialized in `_prepare` instead of constructor

Differential Revision: [D30287179](https://our.internmc.facebook.com/intern/diff/D30287179/)

[ghstack-poisoned]

raghuramank100 commented on torch/ao/sparsity/experimental/pruner/base_pruner.py:
```diff
@@ -15,17 +15,17 @@
 }
 
-def _module_to_path(model, layer, prefix=''):
+def _module_to_fqn(model, layer, prefix=''):
```
This function already exists in https://github.com/pytorch/pytorch/blob/master/torch/ao/sparsity/sparsifier/base_sparsifier.py#L17. Can you move these functions to sparsifier/utils.py and reuse them here?

raghuramank100 commented on torch/ao/sparsity/experimental/pruner/base_pruner.py:
```diff
@@ -53,43 +53,14 @@ class BasePruner(abc.ABC):
 
     """
-    def __init__(self, model, config, defaults):
+    def __init__(self, defaults):
```
Is it possible to make the BasePruner inherit from the Base Sparsifier? This way code for setstate/getstate etc can be shared.

raghuramank100 commented on torch/ao/sparsity/experimental/pruner/base_pruner.py:
```diff
@@ -166,13 +176,18 @@ def convert(self, use_path=False, *args, **kwargs):
 
+    def convert(self):
+        # TODO: Call the torch.ao.utils.convert in here
+        raise NotImplementedError('`convert` is not implemented. Please, use '
```
We should have a different error message here, perhaps just say "convert is not implemented" or "use a different pruner for conversion"

raghuramank100 commented on torch/ao/sparsity/experimental/pruner/base_pruner.py:
```diff
@@ -166,13 +176,18 @@ def convert(self, use_path=False, *args, **kwargs):
+        # TODO: Call the torch.ao.utils.convert in here
+        raise NotImplementedError('`convert` is not implemented. Please, use '
+                                  '`torch.ao.utils.convert` instead.')
+
```
Hoping that most of the methods dont need to be written again and can be just inherited.

raghuramank100 reviewed (changes requested):
Please see comments

kazhou committed:
Update on "[pruner] amend base pruner API to match base sparsifier"


Update base pruner API to match base sparsifier API as defined in D28970960 / PR58955

Changes include:
- `enable_mask_update = True` in `__init__`
- `prepare` takes model and config instead of constructor
- convert functionality renamed to `squash_mask`, `convert` method call now raises Error
- `activation_handles` ad `bias_handles` initialized in `_prepare` instead of constructor

Differential Revision: [D30287179](https://our.internmc.facebook.com/intern/diff/D30287179/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] amend base pruner API to match base sparsifier"


Update base pruner API to match base sparsifier API as defined in D28970960 / PR58955

Changes include:
- `enable_mask_update = True` in `__init__`
- `prepare` takes model and config instead of constructor
- convert functionality renamed to `squash_mask`, `convert` method call now raises Error
- `activation_handles` ad `bias_handles` initialized in `_prepare` instead of constructor

Differential Revision: [D30287179](https://our.internmc.facebook.com/intern/diff/D30287179/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] amend base pruner API to match base sparsifier"


Update base pruner API to match base sparsifier API as defined in D28970960 / PR58955

Changes include:
- `enable_mask_update = True` in `__init__`
- `prepare` takes model and config instead of constructor
- convert functionality renamed to `squash_mask`, `convert` method call now raises Error
- `activation_handles` ad `bias_handles` initialized in `_prepare` instead of constructor

Differential Revision: [D30287179](https://our.internmc.facebook.com/intern/diff/D30287179/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] amend base pruner API to match base sparsifier"


Update base pruner API to match base sparsifier API as defined in D28970960 / PR58955

Changes include:
- `enable_mask_update = True` in `__init__`
- `prepare` takes model and config instead of constructor
- convert functionality renamed to `squash_mask`, `convert` method call now raises Error
- `activation_handles` ad `bias_handles` initialized in `_prepare` instead of constructor

Differential Revision: [D30287179](https://our.internmc.facebook.com/intern/diff/D30287179/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] amend base pruner API to match base sparsifier"


Update base pruner API to match base sparsifier API as defined in D28970960 / PR58955

Changes include:
- `enable_mask_update = True` in `__init__`
- `prepare` takes model and config instead of constructor
- convert functionality renamed to `squash_mask`, `convert` method call now raises Error
- `activation_handles` ad `bias_handles` initialized in `_prepare` instead of constructor

Differential Revision: [D30287179](https://our.internmc.facebook.com/intern/diff/D30287179/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] amend base pruner API to match base sparsifier"


Update base pruner API to match base sparsifier API as defined in D28970960 / PR58955

Changes include:
- `enable_mask_update = True` in `__init__`
- `prepare` takes model and config instead of constructor
- convert functionality renamed to `squash_mask`, `convert` method call now raises Error
- `activation_handles` ad `bias_handles` initialized in `_prepare` instead of constructor

Differential Revision: [D30287179](https://our.internmc.facebook.com/intern/diff/D30287179/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] amend base pruner API to match base sparsifier"


Update base pruner API to match base sparsifier API as defined in D28970960 / PR58955

Changes include:
- `enable_mask_update = True` in `__init__`
- `prepare` takes model and config instead of constructor
- convert functionality renamed to `squash_mask`, `convert` method call now raises Error
- `activation_handles` ad `bias_handles` initialized in `_prepare` instead of constructor

Differential Revision: [D30287179](https://our.internmc.facebook.com/intern/diff/D30287179/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] amend base pruner API to match base sparsifier"


Update base pruner API to match base sparsifier API as defined in D28970960 / PR58955

Changes include:
- `enable_mask_update = True` in `__init__`
- `prepare` takes model and config instead of constructor
- convert functionality renamed to `squash_mask`, `convert` method call now raises Error
- `activation_handles` ad `bias_handles` initialized in `_prepare` instead of constructor

Differential Revision: [D30287179](https://our.internmc.facebook.com/intern/diff/D30287179/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] amend base pruner API to match base sparsifier"


Update base pruner API to match base sparsifier API as defined in D28970960 / PR58955

Changes include:
- `enable_mask_update = True` in `__init__`
- `prepare` takes model and config instead of constructor
- convert functionality renamed to `squash_mask`, `convert` method call now raises Error
- `activation_handles` ad `bias_handles` initialized in `_prepare` instead of constructor

Differential Revision: [D30287179](https://our.internmc.facebook.com/intern/diff/D30287179/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] amend base pruner API to match base sparsifier"


Update base pruner API to match base sparsifier API as defined in D28970960 / PR58955

Changes include:
- `enable_mask_update = True` in `__init__`
- `prepare` takes model and config instead of constructor
- convert functionality renamed to `squash_mask`, `convert` method call now raises Error
- `activation_handles` ad `bias_handles` initialized in `_prepare` instead of constructor

Differential Revision: [D30287179](https://our.internmc.facebook.com/intern/diff/D30287179/)

[ghstack-poisoned]
